### PR TITLE
Implement Admin API to split and merge event processor segments.

### PR DESCRIPTION
**Pull request by saratry**

### CODE REVIEW

File: src/main/java/io/axoniq/axonserver/connector/admin/AdminChannel.java [hunk 1]
```diff
@@ -25,12 +41,36 @@
...
+     * Request to split the biggest segment of a specific event processor.
+     * Returns a {@link CompletableFuture} that completes when the request has been received by AxonServer.
+     * This doesn't imply that the segment has been split already, but only that the request has been properly
+     * delivered.
+     *
+     * @param eventProcessorName   the name of the event processor to split
+     * @param tokenStoreIdentifier the token store identifier of the processor to split
+     * @return a {@link CompletableFuture} that completes when the request has been delivered to AxonServer
+     */
+    CompletableFuture<Void> splitEventProcessor(String eventProcessorName, String tokenStoreIdentifier);
```
[the remark this answers is not in the document]
Comment (smcvb) [reply src/main/java/io/axoniq/axonserver/connector/admin/AdminChannel.java:63]: > I don't think this improvement will require changing the AdminChannel, as a CompletableFuture could complete exceptionally any time the requested operation fails on one of the clients.

Whether not succeeding with a split/merge should be regarded as exceptional is the question.
If the return type is `CompletableFuture<Boolean>`, the result could be a _happy success_, _unhappy success_, and _failure_.

That's at least what Axon Framework has opted for, as you can see [here](https://github.com/AxonFramework/AxonFramework/blob/321cf393de97b013733a044e7c7685caafca98a9/messaging/src/main/java/org/axonframework/eventhandling/StreamingEventProcessor.java?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L90).

File: src/main/java/io/axoniq/axonserver/connector/admin/impl/AdminChannelImpl.java [hunk 2]
```diff
@@ -34,19 +50,38 @@ public AdminChannelImpl(ClientIdentification clientIdentification,
     @Override
     public CompletableFuture<Void> pauseEventProcessor(String eventProcessorName, String tokenStoreIdentifier) {
         EventProcessorIdentifier eventProcessorIdentifier = eventProcessorId(eventProcessorName, tokenStoreIdentifier);
-        CompletableFuture<Void> response = new CompletableFuture<>();
-        CompletableFutureStreamObserver<Empty, Void> responseObserver = new CompletableFutureStreamObserver<>(response);
+        FutureStreamObserver<Empty> responseObserver = new FutureStreamObserver<>(null);
         eventProcessorServiceStub.pauseEventProcessor(eventProcessorIdentifier, responseObserver);
-        return response;
+        return responseObserver.thenAccept(empty -> {
```
[the remark this answers is not in the document]
Comment (smcvb) [reply src/main/java/io/axoniq/axonserver/connector/admin/impl/AdminChannelImpl.java:55]: Honestly, either is ugly :joy: 
So, I go with whatever you prefer.

My gut says a named constant is clear, although it mitigates from what that lambda you give it does.
However, that should be covered in the name.
But, as stated, this is my opinion. To minor a point to not approve this PR for :-)

File: src/test/java/io/axoniq/axonserver/connector/admin/AdminChannelIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
+package io.axoniq.axonserver.connector.admin;
```
[the remark this answers is not in the document]
Comment (smcvb) [reply src/test/java/io/axoniq/axonserver/connector/admin/AdminChannelIntegrationTest.java:17]: Nice!

File: src/main/java/io/axoniq/axonserver/connector/admin/AdminChannel.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (saratry) [reply src/main/java/io/axoniq/axonserver/connector/admin/AdminChannel.java:63]: As we discussed, for this admin API we will keep CompletableFuture<Void>.

[the remark this answers is not in the document]
Comment (smcvb) [reply src/main/java/io/axoniq/axonserver/connector/admin/AdminChannel.java:63]: Agreed!

**Review by smcvb — APPROVED** [review]
My concerns have been addressed, hence approving.

**Pull request merged**